connector/connect/common/src/main/protobuf/spark/connect/expressions.proto

```diff
@@ -405,3 +407,18 @@ message NamedArgumentExpression {
   // (Required) The value expression of the named argument.
   Expression value = 2;
 }
+
+message Origin {
+  // (Required) Indicate the origin type.
+  oneof function {
+    PythonOrigin python_origin = 1;
+  }
+}
+
+message PythonOrigin {
```

Comment:
Please don;t name stuff Python/Scala if it is not language specific.

Reply:
I actually intend this to be language specific. For example, Scala side could have stacktrace chain https://github.com/apache/spark/blob/8bbbde7cb3c396bc369c06853ed3a2ec021a2530/sql/api/src/main/scala/org/apache/spark/sql/catalyst/trees/QueryContexts.scala#L138

This is Python specifically a string for now.

Reply:
I mean, I don't mind combining it for now if you believe it won't be language-specific.

Reply:
Why can't the Python code be modeled as a StackTraceElement? Whats the difference?

Reply:
`StackTraceElement` has JDK dedicated method and fields (e.g., `classLoaderName`). I think we should have a dedicated one for individual languages.

While `DataFrameQueryContext` / `SQLQueryContext` have common information (for now), I think we will end up with having some language-specific and dedicated information for both APIs in the future.

However, I am open to having common one. There is a way to have the common (and, e.g., throw an exception if that information doesn't make sense in some languages).